src/Units/UnitModel.java

```diff
@@ -189,12 +193,40 @@ public boolean canRepairOn(Location locus)
   }
 
   /** Provides a hook for inheritors to supply turn-initialization actions to a unit.
+   * Overriders should collect the output of super.getTurnInitEvents() before returning.
    * @param self Assumed to be a Unit of the model's type.
+   * @param map The current true state of the game map.
    */
   public GameEventQueue getTurnInitEvents(Unit self, MapMaster map)
   {
-    // Most Units don't have any; specific UnitModel types can override.
-    return new GameEventQueue();
+    GameEventQueue queue = new GameEventQueue();
+    if( isAirUnit() && 0 == self.fuel )
```

Comment:
This should check for fuel drain, since sea units also are a thing.

Reply:
Do they die when out of fuel? I guess it make sense since there isn't a way to refuel them, but boats float when not under power.

Reply:
Yup.
I don't make the rules, lol